src/build.c

```diff
@@ -1278,27 +1256,25 @@ static void set_file_error_from_errno(GError **error, gint err, const gchar *pre
 
 
 /* write a little shellscript to call the executable (similar to anjuta_launcher but "internal")
- * fname is the full file name (including path) for the script to create */
-static gboolean build_create_shellscript(const gchar *fname, const gchar *working_dir, const gchar *cmd, gboolean autoclose, GError **error)
+ * it returns the full file name (including path) of the created script */
+static gchar *build_create_shellscript(const gchar *working_dir, const gchar *cmd, gboolean autoclose, GError **error)
 {
-	FILE *fp;
-	gchar *str;
+	gint fd;
+	gchar *str, *fname;
 	gboolean success = TRUE;
 #ifdef G_OS_WIN32
 	gchar *expanded_cmd;
 #else
 	gchar *escaped_dir;
 #endif
+	fd = g_file_open_tmp (RUN_SCRIPT_CMD, &fname, error);
+	if (fd < 0)
+		return NULL;
+	close(fd);
 
-	fp = g_fopen(fname, "w");
-	if (! fp)
-	{
-		set_file_error_from_errno(error, errno, "Failed to create file");
-		return FALSE;
-	}
 #ifdef G_OS_WIN32
 	/* Expand environment variables like %blah%. */
-	expanded_cmd = win32_expand_environment_variables(cmd);
+	expanded_cmd = win32_expand_environment_variables(fname);
 	str = g_strdup_printf("%s\n\n%s\ndel \"%%0\"\n\npause\n", expanded_cmd, (autoclose) ? "" : "pause");
```

Comment:
Interesting, maybe windows first loads it to memory and the file is already closed during its execution (does the 'del' work even when it's at the beginning of the script?).

Seeing some additional temporary files in the project tree is IMO ugly but otherwise it shouldn't be a big problem (unless you start another instance of Geany and want to run something from the same directory in which case I guess the creation of the run script fails).